src/commands/project/init.ts

```diff
@@ -35,20 +36,8 @@ export class ProjectInitCommand extends Command {
   }
 
   async execute(projectDir: string): Promise<void> {
-    const stat = fs.statSync(projectDir, {
-      // returns undefined instead of throwing if the file doesn't exist
-      throwIfNoEntry: false,
-    });
-    if (stat === undefined) {
-      this.error(`No such file or directory ${projectDir}`);
-    } else if (!stat.isDirectory()) {
-      this.error(`Expected a directory, found file at ${projectDir}`);
-    } else {
-      try {
-        fs.accessSync(projectDir, fs.constants.W_OK);
-      } catch (e) {
-        this.error(`Cannot write to directory ${projectDir}`);
-      }
+    if (!isWritableDirectory(projectDir)) {
+      this.error(`${projectDir} must be a writeable directory`);
```

Comment:
`bin/dev schema pull
Error: The project fsl directory: /Users/jdcryans/tests/cli-project/testfsl must be a writeable directory.
    at Object.error (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/errors/index.js:27:15)
    at PullSchemaCommand.error (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/command.js:139:23)
    at PullSchemaCommand.error (/Users/jdcryans/git/fauna-shell/src/lib/fauna-command.js:49:11)
    at PullSchemaCommand.init (/Users/jdcryans/git/fauna-shell/src/lib/schema-command.ts:51:14)
    at async PullSchemaCommand._run (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/command.js:116:13)
    at async Config.runCommand (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/config/config.js:329:25)
    at async Object.run (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/main.js:89:16)`

This is what I get if I delete the folder. Instead we should identify that it's missing.

Reply:
I think updating `isWriteableDirectory` to return `undefined` or a string error might work nicely here, as there's a couple different states the file can be in (missing, is a file, is not writable).

Reply:
wait @jdcryans you get that entire error?
this is what I get
```
fslDir » local-shell schema diff
 ›   Error: The project fsl directory: /Users/chase/fauna/test-schema/fslDir/fsl must be a writeable directory.
```

Reply:
alright updated to be this:
```
fslDir » local-shell schema push                                                                                                                      1 ↵
 ›   Error: The project fsl directory: /Users/chase/fauna/test-schema/fslDir/fsl does not exist.
```